### PR TITLE
Upgrade to curity 3.0

### PR DESCRIPTION
**Pull request by renatoathaydes**
Improved internal build (where access to the Curity's repositories is possible) and better documented pom dependencies.

### CODE REVIEW

**Comment by travisspencer** [comment]
Build fails:

> [ERROR] Failed to execute goal on project identityserver.plugins.authenticators.signicat: Could not resolve dependencies for project io.curity.identityserver.plugin:identityserver.plugins.authenticators.signicat:jar:1.0.0: The following artifacts could not be resolved: se.curity.identityserver:identityserver.sdk:jar:3.0.0, com.signicat:opensaml:jar:1.1-PATCH-6: **Could not find artifact se.curity.identityserver:identityserver.sdk:jar:3.0.0** in central (http://repo.maven.apache.org/maven2)

File: pom.xml [hunk 1]
```diff
@@ -185,5 +199,28 @@
...
+             This will use the stubbed Signicat API provided in the ./java-connector/ directory to be able to
+             compile the source code of this plugin.
+             -->
+        <dependency>
+            <groupId>com.signicat</groupId>
+            <artifactId>java-connector</artifactId>
+            <version>4.0.1</version>
+        </dependency>
+        <dependency>
+            <groupId>com.signicat</groupId>
```
Comment (travisspencer) [inline pom.xml]: this isn't needed. The OpenSAML types that are used are in that "mock" defined above `com.signicat:java-connector:4.0.1`

**Review by travisspencer — CHANGES_REQUESTED** [review]

File: internal-pom.xml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+         xmlns="http://maven.apache.org/POM/4.0.0">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>io.curity.identityserver.plugin</groupId>
+    <artifactId>identityserver.plugins.authenticators.signicat</artifactId>
+    <version>1.0.0</version>
+    <packaging>jar</packaging>
+    <name>Signicat Authenticator</name>
```
Comment (renatoathaydes) [inline internal-pom.xml:26]: This internal POM allows us to build this project without downloading jars manually, as the Signicat jars are installed in the internal Maven repos.
With `mvn package` all the jars we need to install will be placed into `target/libs`.

File: pom.xml [hunk 3]
```diff
@@ -185,5 +201,21 @@
...
+             the README page. -->
+        <dependency>
+            <groupId>com.signicat</groupId>
+            <artifactId>java-connector</artifactId>
+            <version>4.0.1</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-test</artifactId>
```
Comment (renatoathaydes) [inline pom.xml]: but how did this get here!?!?!? IntelliJ magic.

File: README.rst [hunk 4]
```diff
@@ -16,7 +16,11 @@ Curity Identity Server 3.0.0 and `its system requirements <https://developer.cur
 Requirements for Building from Source
 """""""""""""""""""""""""""""""""""""
 
-The source code is written entirely in `Kotlin <http://kotlinlang.org/>`_. It can be compiled using Maven 3. For this to succeed, however, the `Signicat Connector for Java <https://support.signicat.com/display/S2/Signicat+Connector+for+Java>`_ needs to be installed into a Maven repository which is accessible during compilation. The `POM <pom.xml>`_ may need to be updated depending on the Maven Coordinates (Group, Artifact, Version) used during installation. Refer to the `Maven guide for information about installing third-party JARs <https://maven.apache.org/guides/mini/guide-3rd-party-jars-local.html>`_. Once the Signicat Connector's JAR and its associated OpenSAML version are installed, the project can be compiled from a shell by issuing a command like this: ``mvn package``.
+The source code is written entirely in `Kotlin <http://kotlinlang.org/>`_. It can be compiled using Maven 3.
+For this to succeed, however, the `Signicat Connector for Java <https://support.signicat.com/display/S2/Signicat+Connector+for+Java>`_ needs to be installed into a Maven repository which is accessible during compilation.
```
Comment (travisspencer) [inline README.rst:20]: link not fixed

**Review by travisspencer — CHANGES_REQUESTED** [review]

File: README.rst (hunk 4 above)
Comment (renatoathaydes) [reply README.rst:20]: Fixed now.

**Review by travisspencer — APPROVED** [review]

**Pull request merged**